X/XGBoost/build_tarballs.jl

```diff
@@ -57,38 +77,29 @@ fi
 install_license LICENSE
 """
 
-cuda_full_versions = Dict(
-    v"11.0" => v"11.0.3",
-)
-cuda_version = v"11.0"
-
-# These are the platforms we will build for by default, unless further
-# platforms are passed in on the command line
 platforms = expand_cxxstring_abis(supported_platforms())
-cuda_platforms = expand_cxxstring_abis(Platform("x86_64", "linux"; 
-                                        cuda=CUDA.platform(cuda_version)))
 
-for p in cuda_platforms
-    push!(platforms, p)
+cuda_versions_to_build = [v"11.0", v"12.0"]
+
+cuda_platforms = Platform[]
+for cuda_version in cuda_versions_to_build
+    append!(cuda_platforms, expand_cxxstring_abis(Platform("x86_64", "linux"; 
+                            cuda=CUDA.platform(cuda_version))))
 end
 
+append!(platforms, cuda_platforms)
 
 # The products that we will ensure are always built
 products = [
     LibraryProduct(["libxgboost", "xgboost"], :libxgboost),
     ExecutableProduct("xgboost", :xgboost)
 ]
 
-# Dependencies that must be installed before this package can be built
 dependencies = [
     # For OpenMP we use libomp from `LLVMOpenMP_jll` where we use LLVM as compiler (BSD
     # systems), and libgomp from `CompilerSupportLibraries_jll` everywhere else.
     Dependency(PackageSpec(name="CompilerSupportLibraries_jll", uuid="e66e0078-7015-5450-92f7-15fbd957f2ae"); platforms=filter(!Sys.isbsd, platforms)),
     Dependency(PackageSpec(name="LLVMOpenMP_jll", uuid="1d63c593-3942-5779-bab2-d838dc0a180e"); platforms=filter(Sys.isbsd, platforms)),
-
-    # You can only specify one cuda version in the deps. To build against more than 
-    # one cuda version, you have to include them as Archive Sources. (see Torch_jll)
-    BuildDependency(PackageSpec(name="CUDA_full_jll", version=cuda_full_versions[cuda_version]), platforms=cuda_platforms),
```

Comment:
How are you building against CUDA without the toolchain?

Reply:
@maleadt Can you point to an example on how to do this right?

Reply:
Whenever I try to build with more than one `CUDA_full_jll` version, everything runs locally, but then it fails on buildkite.

For example, here is my second commit's build: https://buildkite.com/julialang/yggdrasil/builds/3050#018830ce-3515-460c-947f-f84c6e10133d

All CUDA versions error with this: 
```
[21:48:02] CMake Error at /usr/share/cmake/Modules/CMakeDetermineCUDACompiler.cmake:179 (message):
--
  | [21:48:02]   Failed to find nvcc.
  | [21:48:02]
  | [21:48:02]   Compiler requires the CUDA toolkit.  Please set the CUDAToolkit_ROOT
  | [21:48:02]   variable.
  | [21:48:02] Call Stack (most recent call first):
  | [21:48:02]   CMakeLists.txt:148 (enable_language)
```

I may have just been using the dependencies wrong, so I will revert the changes that put the `CUDA_full_jll` as an ArchiveSource and see if it works.

Reply:
Sorry, missed your comment here. CUDA_full_jll isn't being installed when building for CUDA platforms, e.g. try `julia build_tarballs.jl --verbose x86_64-linux-gnu-libgfortran5-cxx11-cuda+12.0` and see how `CUDA_full_jll` isn't listed as part of the packages that are being installed (or pass `--debug` and see how `nvcc` isn't available). I'm not entirely sure why, but you did structure your `build_tarballs.jl` differently then how other packages do it. See e.g. AMGX, Extrae, PAPI, NVTX, etc.

Reply:
> See e.g. AMGX, Extrae, PAPI, NVTX, etc.

Ah, thank you. I will restructure it to follow the this style.